mlflow/langchain/_langchain_autolog.py

```diff
@@ -180,6 +228,682 @@ def _chain_with_retriever(model):
     return False
 
 
+def get_mlflow_langchain_callback():
+    from langchain.callbacks.base import BaseCallbackHandler
+    from langchain_core.agents import AgentAction, AgentFinish
+    from langchain_core.documents import Document
+    from langchain_core.load import dumpd
+    from langchain_core.messages import BaseMessage
+    from langchain_core.outputs import (
+        ChatGeneration,
+        ChatGenerationChunk,
+        GenerationChunk,
+        LLMResult,
+    )
+    from tenacity import RetryCallState
+
+    from mlflow.entities import SpanEvent, SpanType
+    from mlflow.entities.span_status import SpanStatus
+    from mlflow.entities.trace import Trace
+    from mlflow.entities.trace_status import TraceStatus
+    from mlflow.tracing.types.wrapper import MLflowSpanWrapper
+
+    class MLflowLangchainCallback(BaseCallbackHandler, metaclass=ExceptionSafeAbstractClass):
```

Comment:
Can we have two separate callback for current autologging (metrics/artifacts) and trace? I think we will use the trace callback for served model, but not autologging callback is only used within a run.

Reply:
Splitted!

Reply:
Thanks a lot!!